src/state/ingress_per_unit.py

```diff
@@ -87,4 +89,6 @@ def from_provider(
                     raise IngressPerUnitIntegrationDataValidationError(
                         "Validation of ingress per unit relation data failed."
                     ) from exc
+                except ValueError as exc:
+                    raise IngressPerUnitIntegrationDataValidationError(str(exc)) from exc
```

Comment:
Can you let me know what part of the code is supposed to raise ValueError? iirc pydantic should always wrap it with `ValidationError`